whylog/config/rule.py

```diff
@@ -32,6 +33,29 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def _gather_causes_frequency_information(self):
+        causes_with_frequency_info = [(self._causes[0].name, 0)]
+        for cause in self._causes:
```

Comment:
Did you assume here that self._causes is ordered by cause name? If yes you can't sort because you lost mapping in clues_groups in rule's constraints. Anyway I proposed you: ```from collections import Counter```.